### ISSUE TITLE
I've a problem with the commit command! :)

### ISSUE THREAD

**Issue by ecartolano** [ecartolano/BIOL-653-CARTOLANO-ETIENNE#2]
Hi Jillian!!! 

Have I to use the command "git add" every time before commit?
I don't think... but if I try just to commit an error appears like this one: "not staged for commit"...
But the "add" command isn't to add to stage? I didn't understand.

--------------------------------------------------------------------------------------------------------
MacBook-Air-de-Etienne:jilianclass ecartolano$ git commit -m "My first 5 commit command"
On branch master
Changes not staged for commit:
	modified:   My First History.Rmd

Untracked files:
	.gitignore
	My_First_History.html
	intro_to_r_class1.R
	intro_to_r_class1.html
	jilianclass.Rproj

no changes added to commit
MacBook-Air-de-Etienne:jilianclass ecartolano$ 
 ----------------------------------------------------------------------------------------------------------------------------------
If I try to add before, then I can commit, but, the result doesn't go to the git hub!!! It don't appears at the site!! :)

 ----------------------------------------------------------------------------------------------------------------------------------
MacBook-Air-de-Etienne:jilianclass ecartolano$ git add "My First History.Rmd"
MacBook-Air-de-Etienne:jilianclass ecartolano$ git commit -m "My first 6 commit command"
[master 9014c27] My first 6 commit command
 1 file changed, 1 insertion(+), 1 deletion(-)
MacBook-Air-de-Etienne:jilianclass ecartolano$  
 ----------------------------------------------------------------------------------------------------------------------------------

Please, help me!! :)

**Comment by jdunic** [comment]
You do need to `git add` before each commit, because you need to stage the file (or files) that you want to commit. Which does seem trivial for single file commits, but make sense when working on bigger projects.

You were correct in doing:  
~~~
git add "My First History.Rmd"
~~~

(but note, you'll want to change the filename to a 'My First History.md' so that it will render properly on Github)  

----------------------------------------------------------------------------------------------------------------------------------------

To make your changes appear on Github, you need to *push* your local repository. I see that you've already setup your remote repository. Therefore, you should be able to do

~~~
# make sure you are up to date with what is online
git pull origin master
~~~

~~~
# push your local repo to the remote
git push origin master
~~~

Let me know if that works!

**Comment by ecartolano** [comment]
hummm

I've to use these commands each time...

1 - git add  (to add to staging area)
2 - git commit (to add to repository)
3 - git push (to send the repository to git hub)

so, if I’m working in a team, I’ve to pull the archives each time before work, right?
in this case, the files will be add to my local repository, right?

thank you!!


Etienne CARTOLANO

+1 (617) 310 1944

**Comment by jdunic** [comment]
Yeah, that's a great summary of what to do! Step 3, pushing to the remote repository on Github isn't something that you have to do every time you commit a change. If for some reason you need the updates to be online right away, then you would do the `git push` step. But if not, it's something that you can do at the end of the day or when you'e finished working on some aspect of the project. 

For example, when I make changes to the website, I might make 10 commits over the course of an afternoon, then push my changes to Github when I've decided that I'm ready to update the website. 

You've also got it with the pulling. If you've been working with yourself, possibly on different computers, or are working with collaborators, then you'll want to pull the changes to your local repository.
